rust/worker/src/lib.rs

```diff
@@ -4,7 +4,7 @@ mod config;
 mod execution;
 mod log;
 mod memberlist;
-mod segment;
+pub mod segment;
```

Comment:
unsure if we want to put segment in a separate crate, could use a second opinion
leaning towards keeping it in the worker crate for now

Reply:
would prefer if we didn't have to make it pub from the workers standpoint which suggests separate crate